File: src/main/java/net/adventuregame/registries/ItemRegistry.java
package net.adventuregame.registries;

import net.adventuregame.items.Item;

public class ItemRegistry {
    public static final Registry<Item> REGISTRY = new Registry<>();

    public static void register(String name, Item item) {
        REGISTRY.register(name, item);
    }

    public static Item get(String name) {
        return REGISTRY.get(name);
    }
}

File: src/main/kotlin/net/adventuregame/registries/ItemRegistry.kt
package net.adventuregame.registries

import net.adventuregame.items.Item

object ItemRegistry {
    val REGISTRY: Registry<Item> = Registry()

    fun register(name: String, item: Item) {
        REGISTRY.register(name, item)
    }

    fun get(name: String): Item {
        return REGISTRY.get(name)
    }
}


